lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1613,30 +1624,72 @@ class HSAQueue final : public KalmarQueue
             printAsyncOps(std::cerr);
         }
 
+        std::lock_guard<std::recursive_mutex> lg(qmutex);
+
+#if !defined(NDEBUG)
+        bool foundFirstValidOp = false;
+#endif
+        int first_nullptr = -1;
+        int lastWaitOp = asyncOps.size();
+        if (drainingQueue_) {
+            lastWaitOp = (lastWaitOp * QUEUE_FLUSHING_FRAC);
```

Comment:
nit: maybe a static_cast here?

Reply:
static_cast the asyncOps.size() to int?  Or static_cast `lastWaitOp*QUEUE_FLUSHING_FRAC` to int?

Reply:
static_cast `lastWaitOp * QUEUE_FLUSHING_FRAC`. I don't really like implicit casting for floating point numbers.